lib/chef/knife/core/windows_bootstrap_context.rb

```diff
@@ -238,6 +256,10 @@ def local_download_path
           local_download_path = "%TEMP%\\chef-client-latest.msi"
         end
 
+        def msi_url
+          msi_url = @config[:msi_url] || ''
```

Comment:
how about just ```@config[:msi_url] || ''```? 

Or if the default for @config[:msi_url] was not set, even just ```@config[msi_url]```

or my other comment I guess.